### PR TITLE
chore: fix: don't fail release if commit/push is a no-op (for 7.yy.z releases)

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
release script was failing, when updating image tags in che.properties (in bugfix release script makes no change, so there is nothing to commit)

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: make-release.sh [hunk 1]
```diff
@@ -395,13 +395,16 @@ bumpVersion() {
...
-    git checkout ${BRANCH}
-    cd .ci
-    ./set_tag_version_images.sh ${CHE_VERSION}
-    cd ..
-    git commit -asm "chore: Set ${CHE_VERSION} release image tags"
-    git push origin ${BRANCH}
+    pushd che-server >/dev/null
+        git checkout ${BRANCH}
+        pushd .ci >/dev/null
+            ./set_tag_version_images.sh
```
Comment (nickboldt) [inline make-release.sh]: this doesn't need to have `${CHE_VERSION}` passed in?

File: make-release.sh [hunk 2]
```diff
@@ -395,13 +404,17 @@ bumpVersion() {
...
-    git checkout ${BRANCH}
-    cd .ci
-    ./set_tag_version_images.sh ${CHE_VERSION}
-    cd ..
-    git commit -asm "chore: Set ${CHE_VERSION} release image tags"
-    git push origin ${BRANCH}
+    pushd che-server >/dev/null
+        git checkout ${BRANCH}
+        plugin_version="latest"
+        sed_in_place -r -e "s#che.factory.default_editor=eclipse/che-theia/.*#che.factory.default_editor=eclipse/che-theia/$plugin_version#g" assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties
```
Comment (nickboldt) [inline make-release.sh:410]: this seems wrong since theia is no longer the default editor.

File: make-release.sh (hunk 1 above)
Comment (mkuznyetsov) [reply make-release.sh]: not anymore, the script used to update many entries in che.properties, and probably helm charts, but now it's reduced to two entries which just use the constant "latest" string (and thus, I remove the remains from the script into make-release.sh)

File: make-release.sh (hunk 2 above)
Comment (mkuznyetsov) [reply make-release.sh:410]: I'd leave this as is, is so we could merge the fix for release, and then change it with the rest in a separate issue/PR (probably even this one https://github.com/eclipse-che/che-server/pull/393/files)

**Review by nickboldt — APPROVED** [review]
I would have renamed this issue something like "don't fail if commit or push have nothing to do" as "release hotfix" is pretty non-specific.

**Comment by nickboldt** [comment]
verifying in https://github.com/eclipse-che/che-server/actions/runs/4076761990

**Pull request merged**